philips-hue-adapter.js

```diff
@@ -316,61 +362,91 @@ class PhilipsHueAdapter extends Adapter {
+   * Updates devices known to bridge, instantiating one PhilipsHueDevice per
    * light or updating the existing PhilipsHueDevice
    * @return {Promise}
    */
-  updateLights() {
+  updateDevices() {
     if (!this.username) {
       return Promise.reject('missing username');
     }
 
-    return fetch('http://' + this.bridgeIp + '/api/' + this.username +
-                 '/lights').then(res => {
-      return res.json();
-    }).then(lights => {
+    let apiBase = 'http://' + this.bridgeIp + '/api/' + this.username;
+    return Promise.all([
+      fetch(apiBase + '/lights'),
+      fetch(apiBase + '/sensors')
+    ]).then(responses => {
+      return Promise.all(responses.map(res => res.json()));
+    }).then(([lights, sensors]) => {
       // TODO(hobinjk): dynamically remove lights
-      for (var lightId in lights) {
-        const light = lights[lightId];
-        if (this.lights[lightId]) {
-          // Skip the next update after a sendProperty
-          if (this.recentlyUpdatedLights[lightId]) {
-            delete this.recentlyUpdatedLights[lightId];
-            continue;
-          }
-
-          this.lights[lightId].update(light);
-          continue;
-        }
-        const id = 'philips-hue-' + this.bridgeId + '-' + lightId;
-        this.lights[lightId] = new PhilipsHueDevice(this, id, lightId, light);
+      for (let lightId in lights) {
+        const state = lights[lightId];
+        const deviceId = 'lights/' + lightId;
+        this.updateDevice(deviceId, state);
+      }
+      for (let sensorId in sensors) {
+        const state = sensors[sensorId];
+        const deviceId = 'sensors/' + sensorId;
+        this.updateDevice(deviceId, state);
       }
+
+
       if (this.scheduledUpdate) {
         clearTimeout(this.scheduledUpdate);
       }
-      this.scheduledUpdate = setTimeout(this.updateLights, this.updateInterval);
+      this.scheduledUpdate = setTimeout(this.updateDevices,
+                                        this.updateInterval);
 
     }).catch(e => {
-      console.warn('Error updating lights', e);
+      console.warn('Error updating devices', e);
       if (this.scheduledUpdate) {
         clearTimeout(this.scheduledUpdate);
       }
-      this.scheduledUpdate = setTimeout(this.updateLights, this.updateInterval);
+      this.scheduledUpdate = setTimeout(this.updateDevices,
+                                        this.updateInterval);
     });
   }
 
+  updateDevice(deviceId, deviceState) {
+    let device = this.devices[deviceId];
+    if (device) {
+      if (device.recentlyUpdated) {
+        // Skip the next update after a sendProperty
+        device.recentlyUpdated = false;
+        return;
+      }
+
+      device.update(deviceState);
+      return;
+    }
+
+    if (deviceId.startsWith('sensors')) {
+      if (!SUPPORTED_SENSOR_TYPES[deviceState.type]) {
+        return;
+      }
+    }
+    const normalizedId = deviceId.replace(/lights\//g, '')
```

Comment:
What about `sensors/`?

Reply:
This replacement is to make existing lights' ids not change. Without this replacement the light formerly known as philips-hue-6893894384-1 would be renamed to philips-hue-6893894384-lights-1 which would require anyone who is currently using the adapter to remove the old light and re-add it to update the id.

Reply:
Ah, fair enough.